### DISCUSSION TITLE
Is faster commandline script running possible?

### DISCUSSION THREAD

**Discussion by kortvelyin** [labs4capella/python4capella#229] [Q&A]
Hello,
I have been working on a way to make a communication flow between Unity and Capella. Since running the .py scripts requires the opening and reopening of Capella from the command line every single time, it takes around 10-25 seconds to run the individual scripts. Is there any way to have Capella open in the background or to make the framework setup faster?

**Comment by ylussaud** [comment]
The command line opens Capella and the needed Capella sessions then shutdown Capella. Both the the start of Capella and the session opening take times. Most of the time the script itself will run in few milliseconds or seconds depending of the complexity of the script.
When you keep the Capella opened the code from Python4Capella will only open a session is it's not already opened and never close the session by itself. So the cost of opening the session will only be the first time the session is needed.

One way to keep the Capella running can be to implement a server from a Python4Capella script that would be launched using the command line. This way the Capella will be kept open and you could run needed script through the server socket. If you are working with only one Capella session (project) only the first call would be slower. Or you could open needed sessions at the initialization of the server, this way each call to the server would only cost the time of running the script.

You will find a simple client/server implementation using Python in [this article](https://www.digitalocean.com/community/tutorials/python-socket-programming-server-client). Let me know if you have any issues, didn't test this myself.
